src/Common/FST.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
```

Comment:
Overall if this comes into Common, I'll be pickier on abstractions, etc

Reply:
Actually it can be used for other purposes since it can be treated a compact string:integer dictionary. So it might be better to leave it as is.